Index: open-vm-tools-9.4.6-1770165/configure.ac
===================================================================
--- open-vm-tools-9.4.6-1770165.orig/configure.ac
+++ open-vm-tools-9.4.6-1770165/configure.ac
@@ -937,8 +937,7 @@ if test "$os" = "linux"; then
    CPPFLAGS="$CPPFLAGS -D_FILE_OFFSET_BITS=64"
    CPPFLAGS="$CPPFLAGS -D_LARGEFILE64_SOURCE"
    CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE=500"
-   CPPFLAGS="$CPPFLAGS -D_BSD_SOURCE"
-   CPPFLAGS="$CPPFLAGS -D_SVID_SOURCE"
+   CPPFLAGS="$CPPFLAGS -D_DEFAULT_SOURCE"
 
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
 
